amethyst_assets/src/asset.rs

```diff
@@ -155,15 +166,15 @@ where
     /// The kind of error it may produce.
     type Error: Error;
 
-    /// Returns a list of supported extension (without `.`).
+    /// Returns a list of media types with supported extensions (without `.`).
     ///
     /// ## Examples
     ///
-    /// * `"png"`
-    /// * `"obj"`
-    /// * `"wav"`
-    fn extensions() -> &'static [&'static str];
+    /// * `[("image/png", ["png"), ("image/jpeg", ["jpg", "jpeg"])]`
+    /// * `[("text/plain", ["obj"])]`
+    /// * `[("audio/wav", ["wav"])]`
+    fn media_extensions() -> &'static [(&'static MediaType, &'static [&'static str])];
```

Comment:
Adapted that from the old asset management, where `AssetLoader` for image for example could load png, jpg, tga and bmp. 
https://github.com/amethyst/amethyst/blob/develop/src/asset_manager/asset_manager.rs#L352

We can change that behaviour. But since png, jpg, ... are also the same kind of "image" assets, that would make sense.